subsys/net/ip/l2/openthread/openthread_utils.c

```diff
@@ -0,0 +1,293 @@
+/*
+ * Copyright (c) 2017 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <kernel.h>
+#include <net/net_pkt.h>
+#include <net/openthread.h>
+#include <openthread/openthread.h>
+
+#include "openthread_utils.h"
+
+#if defined(CONFIG_OPENTHREAD_L2_DEBUG_DUMP_15_4) || \
+	defined(CONFIG_OPENTHREAD_L2_DEBUG_DUMP_IPV6)
+void dump_pkt(const char *str, struct net_pkt *pkt)
+{
+	struct net_buf *frag;
+	int n = 0;
+	int i;
+
+	printk("%s packet len %d\r\n", str, net_buf_frags_len(pkt->frags));
+
+	for (frag = pkt->frags; frag; frag = frag->frags) {
+		for (i = 0; i < frag->len; i++) {
+			if (n % 16 == 0) {
+				printk("%s %08X ", str, n);
+			}
+
+			printk("%02X ", frag->data[i]);
+
+			n++;
+			if (n % 8 == 0) {
+				if (n % 16 == 0) {
+					printk("\r\n");
+				} else {
+					printk(" ");
+				}
+			}
+		}
+	}
+
+	if (n % 16) {
+		printk("\r\n");
+	}
+}
+#endif
+
+int pkt_list_add(struct openthread_context *context, struct net_pkt *pkt)
+{
+	u16_t i_idx = context->pkt_list_in_idx;
+
+	if (context->pkt_list_full) {
+		return -ENOMEM;
+	}
+
+	i_idx++;
+
```

Comment:
you can remove that empty line, as the below condition is testing the previous assignment on i_idx.